From 207f1019e70ee88c85bc56ecd2f08bb03caa0eb5 Mon Sep 17 00:00:00 2001
Message-Id: <patch-1.1-207f1019e70-20211021T195940Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Mon, 18 Oct 2021 14:33:13 +0200
Subject: [PATCH] Makefile: remove redundant GIT-CFLAGS dependency from
 "sparse"
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The "sparse" target needed the GIT-CFLAGS dependency before my
c234e8a0ecf (Makefile: make the "sparse" target non-.PHONY,
2021-09-23), but since then it depends on the corresponding *.o files,
which in turn depend on the correct header files, as well as on
GIT-CFLAGS. There's no need to re-state this dependency here.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
A tiny redundancy fix for an already-in-master topic.

 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 381bed2c1d2..12be39ac497 100644
--- a/Makefile
+++ b/Makefile
@@ -2901,7 +2901,7 @@ check-sha1:: t/helper/test-tool$X
 
 SP_OBJ = $(patsubst %.o,%.sp,$(C_OBJ))
 
-$(SP_OBJ): %.sp: %.c %.o GIT-CFLAGS
+$(SP_OBJ): %.sp: %.c %.o
 	$(QUIET_SP)cgcc -no-compile $(ALL_CFLAGS) $(EXTRA_CPPFLAGS) \
 		-Wsparse-error \
 		$(SPARSE_FLAGS) $(SP_EXTRA_FLAGS) $< && \
-- 
2.33.1.1494.g88b39a443e1

